bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/EntityProvider.java

```diff
@@ -29,61 +31,75 @@
 import javax.ws.rs.ext.MessageBodyWriter;
 import javax.ws.rs.ext.Provider;
 
+import org.eclipse.smarthome.io.json.JsonBindingService;
 import org.osgi.service.component.annotations.Component;
-
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
+import org.osgi.service.component.annotations.Reference;
 
 /**
- * JSON reader/writer for Jersey using GSON.
+ * JSON reader/writer for JAX-RS that uses {@link JsonBindingService} to serialize and deserialize between
+ * Java objects and JSON.
  *
- * @author Simon Kaufmann - initial contribution and API
+ * @author Flavio Costa - Initial contribution and API
  *
- * @param <T>
+ * @param <T> Type to be serialized or deserialized.
  */
 @Provider
 @Produces(MediaType.APPLICATION_JSON)
 @Consumes(MediaType.APPLICATION_JSON)
-@Component(immediate = true)
-public class GsonProvider<T> implements MessageBodyReader<T>, MessageBodyWriter<T> {
+@Component(immediate = true, service = { MessageBodyWriter.class, MessageBodyReader.class })
+public class EntityProvider<T> implements MessageBodyWriter<T>, MessageBodyReader<T> {
+
+    /**
+     * Default charset to be used.
+     */
+    protected final Charset CHARSET = StandardCharsets.UTF_8;
 
-    private final Gson gson;
+    /**
+     * Binding layer.
+     */
+    private JsonBindingService<T> binding;
```

Comment:
`binding` alone is a bit misleading in the ESH context because it is taken for external additions which add support for new hardware. Maybe call this `jsonBinding` or `jsonService`.

Reply:
Renamed to `bindingService`.